test/integration/govuk_index/versioning_test.rb

```diff
@@ -0,0 +1,115 @@
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 123, document["_version"]
+
+    version2 = version1.merge(title: "new title", payload_version: 123)
+
+    @processor.process(stub_message_payload(version2))
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 123, document["_version"]
+    assert_equal version1["title"], document["_source"]["title"]
+  end
+
+  def test_should_discard_message_with_earlier_version_than_existing_document
+    version1 = GovukSchemas::RandomExample
+      .for_schema(notification_schema: "specialist_document")
+      .merge_and_validate(payload_version: 123)
+    base_path = version1["base_path"]
+
+    @processor.process(stub_message_payload(version1))
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 123, document["_version"]
+
+    version2 = version1.merge(title: "new title", payload_version: 122)
+
+    @processor.process(stub_message_payload(version2))
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 123, document["_version"]
+    assert_equal version1["title"], document["_source"]["title"]
+  end
+
+  def test_should_delete_and_recreate_document_when_unpublished_and_republished
+    version1 = GovukSchemas::RandomExample
+      .for_schema(notification_schema: "specialist_document")
+      .customise_and_validate({ payload_version: 1 }, ["withdrawn_notice"])
+    base_path = version1["base_path"]
+
+    @processor.process(stub_message_payload(version1))
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 1, document["_version"]
+
+    version2 = version1.merge(payload_version: 2)
+
+    @processor.process(stub_message_payload(version2, unpublishing: true))
+
+    assert_raises(Elasticsearch::Transport::Transport::Errors::NotFound) do
+      fetch_document_from_rummager(id: base_path, index: 'govuk_test')
+    end
+
+    version3 = version1.merge(payload_version: 3)
+
+    @processor.process(stub_message_payload(version3))
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 3, document["_version"]
+  end
+
+  def test_should_discard_unpublishing_message_with_earlier_version
+    version1 = GovukSchemas::RandomExample
+      .for_schema(notification_schema: "specialist_document")
+      .merge_and_validate(payload_version: 2)
+    base_path = version1["base_path"]
+
+    @processor.process(stub_message_payload(version1))
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 2, document["_version"]
+
+    version2 = version1.merge(payload_version: 1)
+
+    @processor.process(stub_message_payload(version2, unpublishing: true))
+
+    document = fetch_document_from_rummager(id: base_path, index: "govuk_test")
+    assert_equal 2, document["_version"]
+  end
+end
```

Comment:
the top 5 lines of each methods are almost identical, might be nice to pull these out to a setup method or similar